src/main/java/filter/expression/Qualifier.java

```diff
@@ -214,16 +192,16 @@ public static void setCurrentTime(LocalDateTime dateTime) {
     }
 
     public boolean isEmptyQualifier() {
-        return name.isEmpty() && content.isPresent() && content.get().isEmpty();
+        return type == QualifierType.EMPTY;
     }
 
     public boolean isFalseQualifier() {
-        return name.equalsIgnoreCase("false") && content.isPresent() && content.get().isEmpty();
+        return type == QualifierType.FALSE;
```

Comment:
That case does happen now if the qualifier is unrecognised. For example, when parsing `a:1`, we look at the 'a' and don't recognise it, so we return `new Qualifier(QualifierType.FALSE, 1)` (previously we would have "a" as the name and it would enter the default switch case, but now we need to use one of the enum values). That results in a false qualifier with content.

I'm actually not sure if we still need `Qualifier.EMPTY`. Previously we weren't able to distinguish the empty qualifier and a qualifier with name "empty", so we had to use an empty content field as well. Now we can, because user input cannot result in `QualifierType.EMPTY`. Maybe it would be better to replace `q == Qualifier.EMPTY` with `q.isEmptyQualifier()` instead?